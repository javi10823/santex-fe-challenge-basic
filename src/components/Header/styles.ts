import styled from 'styled-components';

export const ValueContainer = styled.div({
  padding: 10,
  borderRadius: 5,
  borderWidth: 1,
  borderStyle: 'solid',
  borderColor: 'black',
  backgroundColor: 'white',
});

export const Container = styled.header({
  paddingBlock: 10,
  paddingInline: 20,
  background: 'linear-gradient(to right , #F88484, #F5F5F5)',
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',
});
