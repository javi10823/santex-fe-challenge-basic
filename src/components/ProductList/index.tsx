import { useContext } from 'react';
import { useQuery } from '@apollo/client';
import {
  Typography,
  CardContent,
  CardActions,
  CircularProgress,
  Button,
} from '@mui/material';
import { PRODUCTS } from 'graphql/queries';
import {
  ProductImage,
  LoadingContainer,
  Container,
  ProductCard,
  ErrorContainer,
} from './styles';
import { Context } from 'context/SubtotalContext';
import { Product, Variant } from 'interfaces/products';

export default function ProductList() {
  const { loading, error, data } = useQuery(PRODUCTS());

  const { subtotal, setSubtotal } = useContext(Context);
  const products: Product[] = [];

  if (loading)
    return (
      <LoadingContainer>
        <CircularProgress />
      </LoadingContainer>
    );
  if (error)
    return (
      <ErrorContainer>
        <Typography variant="h4">
          Error: An error occurred while trying to get the products
        </Typography>
      </ErrorContainer>
    );
  data.products.items.map(({ variants }: { variants: Variant[] }) =>
    variants.forEach((variant) => {
      products.push({
        ...variant,
        description: variant.product.description,
        image: variant.product.featuredAsset.source,
      });
    })
  );

  return (
    <Container>
      {products.map((product) => {
        return (
          <ProductCard key={`${product.id}`} sx={{ maxWidth: 345 }}>
            <ProductImage
              component="img"
              src={product?.image}
              alt={product.name}
            />
            <CardContent>
              <Typography gutterBottom variant="h5" component="div">
                {product.name}
              </Typography>
              <Typography variant="body2" color="text.secondary">
                {product.description}
              </Typography>
            </CardContent>
            <CardContent>
              <Typography marginTop={1} variant="h6" color="text.secondary">
                {product.currencyCode} ${product.priceWithTax}
              </Typography>
            </CardContent>
            <CardActions>
              <Button
                type="button"
                size="small"
                onClick={setSubtotal!.bind(
                  null,
                  subtotal! + product.priceWithTax
                )}
              >
                Add to the cart
              </Button>
            </CardActions>
          </ProductCard>
        );
      })}
    </Container>
  );
}
