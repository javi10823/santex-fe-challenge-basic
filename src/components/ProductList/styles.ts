import styled from 'styled-components';
import { CardMedia, CardMediaTypeMap, Card } from '@mui/material';
import { OverridableComponent } from '@mui/material/OverridableComponent';

export const LoadingContainer = styled.div({
  display: 'flex',
  justifyContent: 'center',
  paddingTop: '5%',
});

export const Container = styled.div({
  display: 'flex',
  flexWrap: 'wrap',
  justifyContent: 'center',
});

export const ErrorContainer = styled.div({
  display: 'flex',
  justifyContent: 'center',
  marginTop: 50,
});

export const ProductCard = styled(Card)({
  marginInline: 10,
  marginBlock: 5,
  justifyContent: 'space-between',
  flexDirection: 'column',
  display: 'flex',
});

export const ProductImage = styled(CardMedia)({
  height: 250,
  objectFit: 'contain',
}) as OverridableComponent<CardMediaTypeMap<{}, 'div'>>;
