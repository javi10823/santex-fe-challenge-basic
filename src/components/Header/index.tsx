import { useContext } from 'react';
import { Typography } from '@mui/material';
import { Context } from 'context/SubtotalContext';
import { ValueContainer, Container } from './styles';

export default function Header() {
  const { subtotal } = useContext(Context);

  return (
    <Container>
      <img
        src="https://santex.wpengine.com/wp-content/uploads/2019/02/logo-santex@3x.png"
        alt="logo"
      />
      <ValueContainer>
        <Typography>$ {subtotal}</Typography>
      </ValueContainer>
    </Container>
  );
}
