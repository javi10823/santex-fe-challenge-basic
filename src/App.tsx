import Header from './components/Header';
import ProductList from './components/ProductList';
import SubtotalContext from 'context/SubtotalContext';

function App() {
  return (
    <SubtotalContext>
      <Header />
      <ProductList />
    </SubtotalContext>
  );
}

export default App;
