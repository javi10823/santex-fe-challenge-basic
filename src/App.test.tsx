import { MockedProvider } from '@apollo/client/testing';
import { render, act } from '@testing-library/react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ProductList from './components/ProductList';
import Header from './components/Header';
import App from './App';

describe('Page', () => {
  it('renders home page', async () => {
    const tree = renderer
      .create(
        <MockedProvider mocks={[]} addTypename={false}>
          <App />
        </MockedProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
