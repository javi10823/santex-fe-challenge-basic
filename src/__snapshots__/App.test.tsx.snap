// Jest Snapshot v1, https://goo.gl/fbAQLP


exports[`Page renders home page 1`] = `
Array [
  <header
    className="sc-crzoAE cBTJFm"
  >
    <img
      alt="logo"
      src="https://santex.wpengine.com/wp-content/uploads/2019/02/logo-santex@3x.png"
    />
    <div
      className="sc-iqAclL buaiBc"
    >
      <p
        className="MuiTypography-root-bWTzaV kmUsBR MuiTypography-root MuiTypography-body1"
      >
        $ 
        0
      </p>
    </div>
  </header>,
  <div
    className="sc-dIsUp ccrOdW"
  >
    <span
      className="MuiCircularProgress-root-limSau cuUnHE MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg-kCIYci eYwcxt MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle-cFJhiW kBtAcV MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </span>
  </div>,
]
`;
